src/npctalk.cpp

```diff
@@ -1201,6 +1205,9 @@ std::string dialogue::dynamic_line( const talk_topic &the_topic ) const
     } else if( topic == "TALK_FRIEND_GUARD" ) {
         return _( "I'm on watch." );
 
+    } else if( topic == "TALK_CAMP_OVERSEER" ) {
```

Comment:
Don't hardcode it, use `data/json/npcs/TALK_*.json` files.

Reply:
How were you able to preserve line order when doing this, it loads json responses then hard coded ones?  creating a new json just for TALK_CAMP_OVERSEER is
[
  {
    "id": "TALK_CAMP_OVERSEER",
    "type": "talk_topic",
    "dynamic_line": "Hey Boss...",
    "responses": [
      {
        "text": "See you around.",
        "topic": "TALK_DONE"
      }
    ]
  }
]
I looked at the existing json-ized topics and they still had the companion mission responses hard coded.

Reply:
> How were you able to preserve line order when doing this, it loads json responses then hard coded ones?

That's how it looks in the code. Though adding some other order (such as "put_after") wouldn't be hard.
Responses are added in `json_talk_topic::gen_responses`, which is invoked on top of `dialogue::gen_responses`.

> I looked at the existing json-ized topics and they still had the companion mission responses hard coded.

It could be just because there would be a lot to change.
It's much easier to fix ordering/minor loss of detail than to automatically change the responses to json without breaking something.

Honestly, it would be much better if you modified the json responses system to allow implementing companion missions under them. From what I can tell, vast majority of them only result in text, setting companion mission and calling a specific action.